src/pages/resultsView/coExpression/CoExpressionPlot.tsx

```diff
@@ -0,0 +1,636 @@
+import * as React from "react";
+import {
+    VictoryChart, VictoryAxis, VictoryScatter, VictoryTheme, VictoryLegend, VictoryContainer, VictoryTooltip,
+    VictoryLabel
+} from "victory";
+import {
+    Popover, Overlay
+} from "react-bootstrap";
+import {CoExpression} from "../../../shared/api/generated/CBioPortalAPIInternal";
+import {computed, observable} from "mobx";
+import {observer, Observer} from "mobx-react";
+import {SVGProps} from "react";
+import {MolecularProfile} from "../../../shared/api/generated/CBioPortalAPI";
+import InfoIcon from "../../../shared/components/InfoIcon";
+import {AlterationTypeConstants} from "../ResultsViewPageStore";
+import SvgSaver from "svgsaver";
+import svgToPdfDownload from "shared/lib/svgToPdfDownload";
+import {getSampleViewUrl} from "../../../shared/api/urls";
+import "./styles.scss";
+import Timer = NodeJS.Timer;
+import DefaultTooltip from "../../../shared/components/defaultTooltip/DefaultTooltip";
+import {
+    chartSpaceToDataSpace, computePlotData, getUniquePrecision, invLogScale,
+    logScale
+} from "./CoExpressionPlotUtils";
+import _ from "lodash";
+
+type GeneId = { hugoGeneSymbol:string, entrezGeneId: number, cytoband:string};
+
+export interface ICoExpressionPlotProps {
+    xAxisGene:GeneId;
+    yAxisGene:GeneId;
+    data:{x:number, y:number, mutX:string, mutY:string, studyId:string, sampleId:string}[];
+    coExpression:CoExpression;
+    molecularProfile:MolecularProfile;
+
+    height:number;
+    width:number;
+
+    showLogScaleControls?:boolean;
+    showMutationControls?:boolean;
+    showMutations?:boolean;
+    xLogScale?:boolean;
+    yLogScale?:boolean;
+    handlers:{
+        onClickShowMutations?:()=>void,
+        onClickXLogScale?:()=>void,
+        onClickYLogScale?:()=>void
+    }
+}
+
+enum EventKey {
+    xLogScale="0", yLogScale="1", showMutations="2",// need to be strings because they'll be in input value field
+    downloadSVG="3", downloadPDF="4"
+}
+
+const NO_MUT_STROKE = "#0174df";
+const NO_MUT_FILL = "#58acfa";
+const X_MUT_STROKE = "#886A08";
+const X_MUT_FILL = "#DBA901";
+const Y_MUT_STROKE = "#F7819F";
+const Y_MUT_FILL = "#F5A9F2";
+const BOTH_MUT_STROKE = "#B40404";
+const BOTH_MUT_FILL = "#FF0000";
+
+@observer
+class CoExpressionDataTooltip extends React.Component<{ container:HTMLDivElement, xAxisGene:GeneId, yAxisGene:GeneId, tooltipModel:any, targetHovered:boolean } , {}> {
+    @observable isHovered = false;
+
+    constructor() {
+        super();
+
+        this.onMouseEnter = this.onMouseEnter.bind(this);
+        this.onMouseLeave = this.onMouseLeave.bind(this);
+    }
+
+    private onMouseEnter() {
+        this.isHovered = true;
+    }
+
+    private onMouseLeave() {
+        this.isHovered = false;
+    }
+
+    render() {
+        const {xAxisGene, yAxisGene, tooltipModel, container} = this.props;
+        const arrowOffsetTop = 30; // experimentally determined for aesthetic excellence
```

Comment:
maybe all caps and outside of class?

Reply:
even though its only used in this one place?